### ISSUE TITLE
Performance compared to Mongoose

### ISSUE THREAD

**Issue by zacharynevin** [Bill4Time/mongo-fast-join#8]
I made a minor addition to my initial pull request (small error where `leftKeys.split('.')` should have been `leftKeys[0].split('.')`) and it works perfectly on my system in doing nested joins.

My question is: do you know the performance of this compared to the ad-hoc joins performed in `Mongoose`? I never cared much for ORMs or ODMs, but I was just curious.

**Comment by khous** [comment]
I don't know offhand what the performance difference is. Also sorry about
the delay in accepting the merge request.

**Comment by zacharynevin** [comment]
I notice as well that there is no way to select the columns wanted from the joining collection. I can create a new pull request for this.

**Comment by khous** [comment]
Yeah that's a good idea. A lot of web frameworks support field filtering.

**Comment by zacharynevin** [comment]
Yeah it would be especially useful if you are joining on a user's table, which is probably the most common use case for this.

**Comment by zacharynevin** [comment]
At first glance it looks like I just need to add a projection operator to the `getSubqueries` function. Does this sound correct?

**Comment by khous** [comment]
Not sure. I'm on my phone right now, but yes a projection will work. As
long as the  join fields are returned as well.
